docs/hot-containers.md

```diff
@@ -0,0 +1,102 @@
+# Hot containers
+
+IronFunctions is built on top of container technologies, for each incoming
+workload, it spins a new container, feed it with the payload and sends the
+answer back to the caller. You can expect an average start time of 300ms per
+container. You may refer to [this blog](https://medium.com/travis-on-docker/the-overhead-of-docker-run-f2f06d47c9f3#.96tj75ugb) post to understand the details better.
+
+In the case you need faster start times for your function, you may use a hot
+container instead.
+
+Hot containers are started once and kept alive while there is incoming workload.
+Thus, it means that once you decide to use a hot container, you must be able to
+tell the moment it should reading from standard input to start writing to
+standard output.
+
+Currently, IronFunctions implements a HTTP-like protocol to operate hot
+containers, but instead of communication through a TCP/IP port, it uses standard
+input/output.
+
+## Implementing a hot container
+
+In the [examples directory](https://github.com/iron-io/functions/blob/master/examples/hotcontainers/http/func.go), there is one simple implementation of a hot container
+which we are going to get in the details here.
+
+The basic cycle comprises three steps: read standard input up to a previosly
+known point, process the work, the write the output to stdout with some
+information about when functions daemon should stop reading from stdout.
+
+In the case at hand, we start by plugging stdin to a HTTP request parser:
+
+```go
+r := bufio.NewReader(os.Stdin)
```

Comment:
Shouldn't have a loop somewhere here? Or the idea here really not show that part of the code?